odoo/models.py

```diff
@@ -996,11 +996,16 @@ def to_xid(record_id):
                 for record in self
             )
 
+        nohash = self._context.get('nohash')
         xids.update(
-            (r.id, (modname, '%s_%s_%s' % (
+            (r.id, (modname, nohash and ('%s_%s' % (
```

Comment:
Use f-string please, it is much more readable.

Reply:
Done